backend/fms_core/viewsets/sample.py

```diff
@@ -298,7 +298,7 @@ def retrieve(self, _request, pk=None, *args, **kwargs):
     def list(self, _request, *args, **kwargs):
         samples_queryset = self.filter_queryset(self.get_queryset())
         serialized_data = fetch_sample_data([], samples_queryset, self.request.query_params)
-        return Response(serialized_data)
+        return Response({"results": serialized_data, "count": len(serialized_data)})
```

Comment:
You want the count of the sample_queryset once filtered, not the serialized_data. Otherwise user will always see the first X (depending on limit) items and won't be able to use pagination.